pyuvdata/uvdata/mir_parser.py

```diff
@@ -967,125 +1754,1691 @@ def from_file(
-        # on read
-        self.in_data = self.in_read
-        self.eng_data = self.eng_read
-        self.bl_data = self.bl_read
-        self.sp_data = self.sp_read
-        self.codes_data = self.codes_read
-        self.we_data = self.we_read
+        # it faster to read in the data.
+        file_dict = {}
+        if self._has_cross:
+            int_dict, self._sp_dict = self.sp_data._generate_recpos_dict()
+            file_dict["cross"] = {
+                "int_dict": int_dict,
+                "filetype": "sch_read",
+                "ignore_header": False,
+            }
 
         if self._has_auto:
-            # If the data has auto correlations, then scan the auto file, pull out
-            # the metadata, and get the data index locatinos for faster reads.
-            self.ac_read = self.scan_auto_data(filepath)
-            self.ac_filter = np.ones(self.ac_read.shape, dtype=np.bool_)
-            self.ac_data = self.ac_read
-        else:
-            self.ac_read = None
-            self.ac_filter = None
-            self.ac_data = None
+            filetype = "ach_read"
+            if self.ac_data._old_type:
+                # If we have the old-style file we are working with, then we need to
+                # do two things: first, clean up entries that don't actually have any
+                # data in them (the old format recorded lots of blank data to disk),
+                # and plug in some missing metadata.
+                self._fix_acdata()
+                filetype = "autoCorrelations"
+            int_dict, self._ac_dict = self.ac_data._generate_recpos_dict()
+
+            file_dict["auto"] = {
+                "int_dict": int_dict,
+                "filetype": filetype,
+                "ignore_header": self.ac_data._old_type,
+            }
 
-        # Raw data aren't loaded on start, because the datasets can be huge
-        # You can force this after creating the object with load_data().
-        self.vis_data = None
-        self.raw_data = None
-        self.auto_data = None
-        self.raw_scale_fac = None
-
-        # These dicts match index number to array position. It ends up being filled
-        # by _update_filter, which is called inside of load_data below.
-        self.inhid_dict = {}
-        self.blhid_dict = {}
-        self.sphid_dict = {}
-
-        # If requested, now we load out the visibilities.
-        self.load_data(
-            load_vis=load_vis,
-            load_raw=load_raw,
-            load_auto=(load_auto and self._has_auto),
-        )
+        self._file_dict = {filepath: file_dict}
+        self.filepath = filepath
 
-    def __init__(
+        # Set/clear these to start
+        self.vis_data = self.raw_data = self.auto_data = None
+        self._tsys_applied = False
+
+        # If requested, now we load up the visibilities.
+        self.load_data(load_vis=load_vis, load_raw=load_raw, load_auto=load_auto)
+
+    def tofile(
```

Comment:
I notice you're using "tofile" and "fromfile" rather than "read" and "write" throughout. That's different than what we use on pyuvdata, which might possibly be confusing for users. I don't have strong feelings on this front, but wanted to point it out.

Reply:
So that was somewhat inherited from numpy's read/write patterns, which originally was being used to read/write these attributes to/from disk (as they were just ndarrays). But now that this has evolved a little bit, I've renamed things to match the more typical usage pattern of `read` and `write` (in a2f46b4b).